improver/spotdata/spot_extraction.py

```diff
@@ -195,27 +214,20 @@ def process(self, neighbour_cube, diagnostic_cube, new_title=None):
 
         coordinate_cube = self.extract_coordinates(neighbour_cube)
 
-        # Deal with leading dimensions such as thresholds, realizations, etc.
-        data_cubes = iris.cube.CubeList()
-        for cube in diagnostic_cube.slices(
-            [
-                diagnostic_cube.coord(axis="x").name(),
-                diagnostic_cube.coord(axis="y").name(),
-            ]
-        ):
-
-            spot_values = self.extract_diagnostic_data(coordinate_cube, cube)
-            spotdata_cube = self.build_diagnostic_cube(
-                neighbour_cube, cube, spot_values
-            )
+        spot_values = self.extract_diagnostic_data(coordinate_cube, diagnostic_cube)
 
-            # Add scalar coordinates onto the spot cube which can be promoted
-            # to reform and leading dimensions.
-            for coord in cube.coords(dim_coords=False):
-                spotdata_cube.add_aux_coord(coord)
-            data_cubes.append(spotdata_cube)
+        additional_dims = None
+        if len(spot_values.shape) > 1:
+            additional_dims = np.flip([dim for dim in diagnostic_cube.dim_coords])[2:]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of a comprehension](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5657052)